packages/page-staking/src/Performance/useCommitteePerformance.tsx

```diff
@@ -0,0 +1,272 @@
+// Copyright 2017-2022 @polkadot/app-staking authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+import { createNamedHook, useApi } from '@polkadot/react-hooks';
+import { StorageKey } from '@polkadot/types';
+import { Hash } from '@polkadot/types/interfaces';
+import { AnyTuple, Codec } from '@polkadot/types/types';
+
+export interface ValidatorPerformance {
+  accountId: string,
+  blockCount?: number,
+  expectedBlockCount: number,
+}
+
+export interface SessionCommitteePerformance {
+  sessionId: number,
+  isPalletElectionsSupported: boolean | undefined,
+  performance: ValidatorPerformance[],
+  firstSessionBlockAuthor: string | undefined,
+}
+
+export function parseSessionBlockCount (sessionValidatorBlockCountValue: [StorageKey<AnyTuple>, Codec][], firstSessionBlockAuthor: string): [string, number][] {
+  return sessionValidatorBlockCountValue.map(([key, values]) => {
+    const account = key.args[0].toString();
+    let count = Number(values.toString());
+
+    if (account === firstSessionBlockAuthor) {
+      // a workaround for the fact that the first session block author is not reflected in that block
+      // elections.sessionValidatorBlockCount state
+      count += 1;
+    }
+
+    return [account, count];
+  });
+}
+
+function useSessionCommitteePerformanceImpl (sessions: number[]): SessionCommitteePerformance[] {
+  const { api } = useApi();
+
+  const MAX_POSSIBLE_SESSIONS = 1000;
+
+  const [firstBlockInSessionHashes, setFirstBlockInSessionHashes] = useState<(Hash | undefined)[]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
```

Comment:
I feel like it would be much cleaner if we would have instead of arrays either dictionary from session to item or arrays of tuples (session, item). Also why we initialize every array to max size at the start rather than create it as an empty list and then based on sessions array?